File: backend/app/schemas/daily_sign.py
from datetime import date

from pydantic import BaseModel, Field


class DailySignRead(BaseModel):
    id: int
    sign_no: int
    title: str
    original_text: str
    plain_explanation: str
    keywords: list[str]
    cultural_source: str
    inspiration: str

    model_config = {"from_attributes": True}


class DrawSignRequest(BaseModel):
    user_key: str = Field(min_length=1, max_length=80)
    question: str | None = Field(default=None, max_length=500)


class TodaySignResponse(BaseModel):
    draw_date: date
    sign: DailySignRead
    note: str = "以下内容为传统文化解读与自我启发，不代表现实预测。"

File: frontend/src/api/dailySign.ts
import { fetchWithAuth } from "./fetchWithAuth";

export type DailySign = {
  id: number;
  sign_no: number;
  title: string;
  original_text: string;
  plain_explanation: string;
  keywords: string[];
  cultural_source: string;
  inspiration: string;
};

export type TodaySignResponse = {
  draw_date: string;
  sign: DailySign;
  note: string;
};

export async function drawTodaySign(userKey: string): Promise<TodaySignResponse> {
  const response = await fetchWithAuth("/daily-signs/draw", {
    method: "POST",
    headers: {
      "Content-Type": "application/json"
    },
    body: JSON.stringify({ user_key: userKey })
  });

  if (!response.ok) {
    const message = await response.text();
    throw new Error(message || "抽签失败，请稍后再试");
  }

  return response.json();
}
